app/pods/components/player-code-challenge/component.js

```diff
@@ -116,14 +117,40 @@ export default class CodeChallengeComponent extends Component {
   @restartableTask
   *runCodeTask (config) {
```

Comment:
Since, this will resolve sooner, the button's state must be bound to something other than `runCodeTask.isRunning`
The button should show loading while we poll for submission

Reply:
Marking as resolved because this works as intended.